File: FrontEnd/controle-estoque/src/types/UsuarioRequest.ts
export interface UsuarioRequest {
  nome: string
  email: string
  senha: string
  telefone: string
  cpf: string
  endereco: {
    cep: string
    rua: string
    numero: string
    bairro: string
    cidade: string
    estado: string
  }
  role: string
}

File: Backend/src/main/java/br/com/estoque/dto/UsuarioRequest.java
package br.com.estoque.dto;

import br.com.estoque.model.enums.Role;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class UsuarioRequest {

    @NotBlank
    private String nome;

    @Email
    @NotBlank
    private String email;

    @NotBlank
    private String senha;

    @NotBlank
    private String telefone;

    @NotBlank
    private String cpf;

    @NotNull
    private Role role;

    private EnderecoDTO endereco;

    @Data
    public static class EnderecoDTO {
        private String cep;
        private String rua;
        private String numero;
        private String bairro;
        private String cidade;
        private String estado;
    }
}
